dev_scripts_helpers/github/sync_gh_issue_labels.py

```diff
@@ -0,0 +1,244 @@
+    Save labels to the label inventory manifest file.
+
+    :param labels: label objects
+    :param path: path to save the label inventory manifest file to
+    """
+    try:
+        with open(path, "w") as file:
+            labels_data = [label.to_dict() for label in labels]
+            # Set `default_flow_style=False` to use block style instead of
+            # flow style for better readability.
+            yaml.dump(
+                labels_data, file, default_flow_style=False, sort_keys=False
+            )
+    except Exception as e:
+        _LOG.error("Error saving label inventory manifest file: %s", str(e))
+        raise e
+
+
+def _parse() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(
+        description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter
+    )
+    hparser.add_verbosity_arg(parser)
+    parser.add_argument(
+        "--yaml",
+        "-y",
+        required=True,
+        help="Path to label inventory manifest file",
+    )
+    parser.add_argument(
+        "--owner",
+        "-o",
+        required=True,
+        help="GitHub repository owner/organization",
+    )
+    parser.add_argument(
+        "--repo", "-r", required=True, help="GitHub repository name"
+    )
+    parser.add_argument(
+        "--token", "-t", required=True, help="GitHub personal access token"
+    )
+    parser.add_argument(
+        "--prune",
+        "-p",
+        action="store_true",
+        help="Delete labels that exist in the repo but not in the label inventory manifest file",
+    )
+    parser.add_argument(
+        "--dry_run",
+        "-d",
+        action="store_true",
+        help="Print out the labels and exits immediately",
+    )
+    parser.add_argument(
+        "--backup",
+        "-b",
+        action="store_true",
+        help="Backup current labels to a label inventory manifest file",
+    )
+    return parser
+
+
+def _main(parser: argparse.ArgumentParser) -> None:
+    args = parser.parse_args()
+    hdbg.init_logger(verbosity=args.log_level, use_exec_path=True)
+    # Load labels from label inventory manifest file.
+    labels = _load_labels(args.yaml)
+    labels_map = {label.name: label for label in labels}
+    # Initialize GH client.
+    client = github.Github(args.token)
+    repo = client.get_repo(f"{args.owner}/{args.repo}")
+    # Build maps for efficient lookup.
+    current_labels = repo.get_labels()
+    current_labels_data = []
+    for label in current_labels:
+        current_labels_data.append(
+            Label(
+                name=label.name,
+                description=label.description if label.description else None,
+                color=label.color,
+            )
+        )
+    current_labels_map = {label.name: label for label in current_labels_data}
+    # Save the labels if backup is enabled.
+    if args.backup:
+        root_dir = hgit.get_client_root(False)
+        dst_dir = f"{root_dir}/dev_scripts_helpers/github/labels/backup"
```

Comment:
I would just do it here in a dir `tmp.backup_sync_gh_issue_labels.yaml`?

We use `tmp` for everything that doesn't need to be checked in